### ISSUE TITLE
ConfigCat - ERROR - [1103] Unexpected error occurred while trying to fetch config JSON. ReferenceError: XMLHttpRequest is not defined

### ISSUE THREAD

**Issue by shadoath** [configcat/react-sdk#40]
### Describe the bug

Each time I run an API route for Next-JS I get this error showing up.

```
ReferenceError: XMLHttpRequest is not defined
    at /code/app/node_modules/configcat-react/lib/cjs/ConfigFetcher.js:33:37
    at new Promise (<anonymous>)
    at HttpConfigFetcher.fetchLogic (/code/app/node_modules/configcat-react/lib/cjs/ConfigFetcher.js:30:16)
    at AutoPollConfigService.<anonymous> (/code/app/node_modules/configcat-common/lib/ConfigServiceBase.js:202:65)
    at step (/code/app/node_modules/tslib/tslib.js:195:27)
    at Object.next (/code/app/node_modules/tslib/tslib.js:176:57)
    at /code/app/node_modules/tslib/tslib.js:169:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/code/app/node_modules/tslib/tslib.js:165:16)
    at ConfigServiceBase.fetchRequestAsync (/code/app/node_modules/configcat-common/lib/ConfigServiceBase.js:191:24)
    at AutoPollConfigService.<anonymous> (/code/app/node_modules/configcat-common/lib/ConfigServiceBase.js:146:51)
    at step (/code/app/node_modules/tslib/tslib.js:195:27)
    at Object.next (/code/app/node_modules/tslib/tslib.js:176:57)
    at /code/app/node_modules/tslib/tslib.js:169:75
    at new Promise (<anonymous>)
```

I have the `<ConfigCatProvider>` wrapping my app inside the Authenticated user context. 

I'm thinking it is related to severside/client side node issues. [This](https://stackoverflow.com/a/32604544/1418337) stack overflow question seemed to make sense.

System:
- configcat-react: 3.1.1
- Next JS 13.5.4
- Node 18.16.0

### Expected behavior

There is no error or a way to denote that this might render on the server side.

**Comment by laliconfigcat** [comment]
Hello @shadoath ,

Currently, the React SDK only supports browser-based usage. If you want to use ConfigCat in a SSR application, I can recommend you to check our SSR SDK: https://configcat.com/docs/sdk-reference/js-ssr/.

Other than this, we have a ticket on our backlog to support SSR scenarios with our React SDK.

Cheers,

**Comment by shadoath** [comment]
Thank you for the update @laliconfigcat. I have switched to the js-ssr version of config cat. 
It would be great to know when the ticket for support to SSR is completed.

**Comment by laliconfigcat** [comment]
I hope the SSR SDK will suit your needs till we finish the development in our React SDK. 
We'll notify you here if we made any progress with supporting SSR with the React SDK.

**Comment by adams85** [comment]
@shadoath FYI, the fix for this issue has been released in [v4.5.0](https://github.com/configcat/react-sdk/releases/tag/v4.5.0).

**Comment by shadoath** [comment]
Thank you!
